Allow to edit content - first pass

jitendraharpalani58 opened the pull request:
See #93 

@danielbachhuber This is what I was thinking.

jitendraharpalani58 committed:
Allow to edit content - first pass

jitendraharpalani58 committed:
refactor "content" to "shortcode_content"

danielbachhuber commented on inc/class-shortcode-ui.php:
```diff
@@ -35,9 +35,10 @@ private function setup_actions() {
+			'label'             => '',
+			'attrs'             => array(),
+			'listItemImage'     => '',   // src or 'dashicons-' - used in insert list.
+			'shortcode_content' => true,
```
Let's call this `inner_content`. I think `shortcode_` is implied.

jitendraharpalani58 replied:
will do

danielbachhuber commented on inc/templates/edit-form.tpl.php:
```diff
@@ -134,3 +134,16 @@
+		</div>
+		<div class="content-edit-container" style="display: none;">
+			<textarea class="content-edit">{{ data.shortcode_content }}</textarea>
+			<a href="#" class="button-save">Save Content</a>
```
Why do you have "Edit" and "Save"?

jitendraharpalani58 replied:
I think it makes sense to edit content, if the need be, directly from shortcake UI.

danielbachhuber replied:
Sure — why not just dump it in a textarea then?

jitendraharpalani58 replied:
Looks more presentable.

mattheu replied:
We don't use the edit/save UX for other fields. I'm not sure that inner content should be different. I think it would be nicer if it just live updated like the other fields.

We can probably re-use the same views we use for other fields. It might be nice if it was customizable too - as a textarea might not always be appropriate.

jitendraharpalani58 replied:
@mattheu Can you think of any use cases where `inner_content` will not be plain text (textarea), neither a combination of nested shortcodes ?

mattheu replied:
Not really - but I could perhaps see that either a simpler text input or a more complex TinyMCE editor might be useful.

jitendraharpalani58 committed:
refactor 'shortcode_content'

remove functionality to edit/save content to keep it consistent with
other fields

jitendraharpalani58 committed:
Merge branch 'master' into 93-allow-content-edit

jitendraharpalani58 committed:
fix few errors caused by refactoring

mattheu commented on js/shortcode-ui.js:
```diff
@@ -837,11 +868,11 @@ var Shortcode_UI;
+				
+				if ( 'content' in options.shortcode ) {
+					shortcode.set( 'content', options.shortcode.content )
+				}
```
I think this should be:

``` php
if ( 'inner_content' in options.shortcode ) {
    shortcode.set( 'inner_content', options.shortcode.content )
}
```

jitendraharpalani58 committed:
save inner_content

jitendraharpalani58 committed:
Default inner content to false

danielbachhuber committed:
Merge branch 'master' into 93-allow-content-edit

danielbachhuber committed:
Make "Inner Content" string translatable

jitendraharpalani58 committed:
broken state - get feedback

jitendraharpalani58 committed:
create model for inner_content

everything working fine now.
fixes all the issues.

danielbachhuber commented on inc/class-shortcode-ui.php:
```diff
@@ -35,13 +35,44 @@ private function setup_actions() {
+			);
+		}
+		
+		$num_attrs = count( $args['attrs'] );
```
Can we have some mention that this is a backwards compatibility shim that we'll be removing in the next version?

jitendraharpalani58 replied:
adding that now.

danielbachhuber commented on inc/templates/edit-form.tpl.php:
```diff
@@ -52,7 +52,9 @@
 	<div class="field-block">
 		<label for="{{ data.attr }}">{{ data.label }}</label>
-		<textarea name="{{ data.attr }}" id="{{ data.attr }}" placeholder="{{ data.placeholder }}">{{ data.value }}</textarea>
+		<textarea name="<#if ( data.attr ) { #>{{ data.attr }}<#}else{#>inner_content<#}#>" 
```
Hm, does this mean I couldn't have `inner_content` as an attribute? Maybe we do need a separate template then.

jitendraharpalani58 replied:
the whole purpose of adding `inner_content` is to not use it as attribute. That was the reason i added a new template for `inner_content`. Should I revert back to adding a new template ?

danielbachhuber replied:
> Should I revert back to adding a new template ?

Yes please. Apologies for the misdirection

danielbachhuber commented on js/shortcode-ui.js:
```diff
@@ -118,7 +118,19 @@ var Shortcode_UI;
+	 */
+	sui.models.InnerContent = Backbone.Model.extend({
+		defaults: {
+			label:       'Inner Content',
```
This default value should be our translatable string.

jitendraharpalani58 replied:
then, I can just leave it empty in JS?

jitendraharpalani58 committed:
add new template, msg for backward compatibility

jitendraharpalani58 committed:
Merge with Master

danielbachhuber committed:
Inspect whether this is defined
